helix-core/src/main/java/org/apache/helix/controller/stages/CurrentStateComputationStage.java

```diff
@@ -236,6 +267,11 @@ private void updateCurrentStates(LiveInstance instance, Collection<CurrentState>
     }
   }
 
+  private void setStaleMessage(CurrentStateOutput currentStateOutput, String instanceName,
```

Comment:
Can we just call "currentStateOutput.setStaleMessage(instanceName, message);" in the caller so we don't need this method?

Reply:
currentStateOutput is already updated with latest stale messages, and accordingly, the cache is updated. So this method is actually redundant, and that's why I used to remove it. But I think from your previous comment, you'd like to an explicit `set` operation?

Reply:
You mean we can just remove it? The comment here is just saying that this private method is not necessary : )

But regarding the set, yes, I think we need explicit set. Implicit set always introduces bugs.

Reply:
I mean this two lines:
updatePendingMessages(instance, messages.values(), relayMessages.values(), currentStateOutput,
          resourceMap);
      cache.setStaleMessages(currentStateOutput.getStaleMessageMap());

The second one is redundant. currentStateOutput is already updated in the function. But seems you think it's better to have the second one?

Reply:
This is related to my other comment. In general, we shall not pass a modifiable map ref directly. Especially if the passed ref will be used to update. And given that you make a shallow copy instead of just pass the ref, then we will need both.